### PR TITLE
Fixes #1459: UVL Reader now uses parent folder even with trivial relative path

### LINKED ISSUE #1459: Feature model manager cannot load UVL models with simple relative file paths
(FeatureIDE library v3.10.0)

When using the feature model manager to load a UVL file, ...
```java
featureModel = FeatureModelManager.load(inputPath);
```
... an exception is thrown when the input path is relative and does not contain a directory component:

```
$ java -jar bin/io.jar x.uvl
ERROR: Cannot invoke "java.nio.file.Path.toString()" because the return value of "java.nio.file.Path.getParent()" is null
java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.toString()" because the return value of "java.nio.file.Path.getParent()" is null
	at de.ovgu.featureide.fm.core.io.uvl.UVLFeatureModelFormat.read(UVLFeatureModelFormat.java:151)
	at de.ovgu.featureide.fm.core.io.uvl.UVLFeatureModelFormat.read(UVLFeatureModelFormat.java:83)
	at de.ovgu.featureide.fm.core.io.manager.SimpleFileHandler.parse(SimpleFileHandler.java:252)
	at de.ovgu.featureide.fm.core.io.manager.AbstractIO.getFileHandler(AbstractIO.java:119)
	at de.ovgu.featureide.fm.core.io.manager.AbstractIO.load(AbstractIO.java:89)
	at de.ovgu.featureide.fm.core.io.manager.FeatureModelManager.load(FeatureModelManager.java:66)
	at Main.main(Main.java:45)
Exception in thread "main" java.lang.RuntimeException: failed to load feature model
	at Main.main(Main.java:57)
```

workaround: augment input path to read `./x.uvl` instead of `x.uvl`, like this:
```java
if (!inputPath.toFile().isAbsolute()) {
  inputPath = Paths.get(".").resolve(inputPath);
}
featureModel = FeatureModelManager.load(inputPath);
```

### PR DESCRIPTION
**Pull request by SundermannC**

### CODE REVIEW

File: plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/uvl/UVLFeatureModelFormat.java [hunk 1]
```diff
@@ -148,7 +149,7 @@ public ProblemList read(IFeatureModel fm, CharSequence source, Path path) {
 		pl = new ProblemList();
 		final UVLModelFactory uvlModelFactory = new UVLModelFactory();
 		try {
-			rootModel = uvlModelFactory.parse(source.toString(), path.getParent().toString());
+			rootModel = uvlModelFactory.parse(source.toString(), getParentDirectoryName(path));
```
Comment (skrieter) [inline plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/uvl/UVLFeatureModelFormat.java:152]: `rootModel = uvlModelFactory.parse(source.toString(), path.resolve("..").normalize().toString()); `

File: plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/uvl/UVLFeatureModelFormat.java [hunk 2]
```diff
@@ -164,6 +165,13 @@ public ProblemList read(IFeatureModel fm, CharSequence source, Path path) {
 		return pl;
 	}
 
+	private String getParentDirectoryName(Path path) {
+		if (path.getParent() == null) { // If the parent directory is not specified just use ./ as parent
+			path = Paths.get(".").resolve(path);
+		}
+		return path.getParent().toString();
+	}
+
```
Comment (skrieter) [inline plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/uvl/UVLFeatureModelFormat.java:174]: remove

File: plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/uvl/UVLFeatureModelFormat.java [hunk 3]
```diff
@@ -22,6 +22,7 @@
 
 import java.io.File;
 import java.nio.file.Path;
+import java.nio.file.Paths;
```
Comment (skrieter) [inline plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/uvl/UVLFeatureModelFormat.java:25]: remove

**Review by skrieter — CHANGES_REQUESTED** [review]

**Comment by SundermannC** [comment]
Embedded feedback, refactoring seems to behave as expected.

**Pull request merged**